spring-cloud-skipper/src/main/java/org/springframework/cloud/skipper/support/PropertiesDiff.java

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.skipper.support;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.apache.commons.lang3.builder.Diff;
+import org.apache.commons.lang3.builder.DiffBuilder;
+import org.apache.commons.lang3.builder.DiffResult;
+import org.apache.commons.lang3.builder.ToStringStyle;
+
+/**
+ * {@code PropertiesDiff} is an implementation to come up with a difference
+ * between two properties sets. This difference gives answer to questions like
+ * what has been added, removed and changed between left and right hand properties.
+ *
+ * @author Janne Valkealahti
+ *
+ */
+public final class PropertiesDiff {
+
+	private Map<String, String> removed = new HashMap<>();
+	private Map<String, String> added = new HashMap<>();
+	private Map<String, String> common = new HashMap<>();
+	private Map<String, PropertyChange> changed = new HashMap<>();
+
+	private PropertiesDiff(Map<String, String> removed, Map<String, String> added, Map<String, String> common,
+			Map<String, PropertyChange> changed) {
+		this.removed = removed;
+		this.added = added;
+		this.common = common;
+		this.changed = changed;
+	}
+
+	/**
+	 * Gets the removed properties.
+	 *
+	 * @return the removed
+	 */
+	public Map<String, String> getRemoved() {
```

Comment:
I'll merge, but interested to know if you think a name more along the lines of 'entriesOnlyOnLeft` would be more consistent since the builder uses 'left' and 'right' methods, this way the terminology stays the same.